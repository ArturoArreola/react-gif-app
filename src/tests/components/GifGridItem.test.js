import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test <GifGridItem/>', () => { 

    const title = 'Gif Title';
    const url = 'http://localhost/mygif.gif';
    const wrapper = shallow(<GifGridItem title={title} url={url} /> )
    
    test('Show the component', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('<p> tag must have title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('<img/> tag must have src and alt values', () => {
        const img = wrapper.find('img');
        console.log(img.props().src);
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    });
    
    test('<div> contains class animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__fadeIn')).toBeTruthy();
    });
 });