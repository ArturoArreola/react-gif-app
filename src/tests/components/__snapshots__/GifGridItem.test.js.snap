// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem/> Show the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Gif Title"
    src="http://localhost/mygif.gif"
  />
  <p>
    Gif Title
  </p>
</div>
`;
